src/EditorFeatures/Core/LanguageServer/Handlers/Symbols/DocumentSymbolsHandler.cs

```diff
@@ -53,7 +54,8 @@ public async Task<object[]> HandleRequestAsync(DocumentSymbolParams request, Req
             // TODO - Return more than 2 levels of symbols.
             // https://github.com/dotnet/roslyn/projects/45#card-20033869
             using var _ = ArrayBuilder<object>.GetInstance(out var symbols);
-            if (context.ClientCapabilities?.TextDocument?.DocumentSymbol?.HierarchicalDocumentSymbolSupport == true)
+            if (context.ClientCapabilities?.TextDocument?.DocumentSymbol?.HierarchicalDocumentSymbolSupport == true
+                || context.ServerKind == WellKnownLspServerKinds.AlwaysActiveVSLspServer)
```

Comment:
This addition is a quick workaround that will be updated (since there isn't a way to set `HierarchicalDocumentSymbolSupport = true` as a parameter when making the LSP request). This is needed so that I'm given `DocumentSymbol[]` instead of `SymbolInformation`.

Reply:
@dibarbet  for ideas

Reply:
this was my suggestion - the client unfortunately does not support hierarchical document symbols and we can't contribute to client capabilities as the extension. 

A solution we should do later (def not should not block this PR) is to pass in a special parameter object that indicates if it should use hierarchical symbols.
e.g.
```
RoslynDocumentSymbolParams : DocumentSymbolParams
{
    public bool? UseHierarchicalSymbols;
}
```